R/labs_to_levels.R

```diff
@@ -0,0 +1,116 @@
+#' @title Convert value labels into factor levels
+#' @name labs_to_levels
+#'
+#' @details
+#' Convert variables or data into factors and uses value labels as factor levels.
+#' This function is similar to `to_factor()`, however, it is provides additional
+#' options for factors with value labels attributes. While `to_factor()` returns
+#' factors as is, `labs_to_levels()` always uses value labels as factor levels,
+#' even for factors.
+#'
+#' @param x A data frame or vector.
+#' @param remove_attr Logical, if `"labels"` attribute should be removed after
+#' value labels are set as factor levels.
+#' @param ... Arguments passed to or from other methods, in this in particular
+#' case to `to_factor()`.
+#' @inheritParams find_columns
+#' @inheritParams categorize
+#'
+#' @return A factor, or a data frame of factors.
+#'
+#' @examples
+#' str(labs_to_levels(iris))
+#'
+#' # use labels as levels
+#' data(efc)
+#' # create factor
+#' x <- as.factor(efc$c172code)
+#' # add value labels - these are not factor levels yet
+#' x <- add_labs(x, values = c(`1` = "low", `2` = "mid", `3` = "high"))
+#' levels(x)
+#' data_tabulate(x)
+#'
+#' x <- labs_to_levels(x)
+#' levels(x)
+#' data_tabulate(x)
+#' @export
+labs_to_levels <- function(x, ...) {
```

Comment:
Let's not abbreviate. `labels_to_levels()` is a more informative name. Saving two characters is not worth the user confusion.

Reply:
These are `labs` for me 😅 

![image](https://user-images.githubusercontent.com/11330453/226939049-413121de-90ec-46b2-891e-b5d987b2303c.png)

Reply:
You can discuss this with @etiennebacher ;-)

Reply:
> Sorry, I missed this. But I'd change this to add_labels() as well.

So what about name conflict with `sjlabelled::add_labels()`? There are a bunch of propositions in the linked comment if you wanna see alternatives

Reply:
Of the remaining suggestions, I like apply_labels() and assign_labels()

Reply:
Yes, both sound good to me

Reply:
Honestly I find both less natural than `add_labs()` but if I had to choose I'd go with `assign_labels()`. What about reexporting the function from `sjlabelled` as @IndrajeetPatil proposed? Is this an option?

Reply:
But then we would need to import _sjlabelled_, or am I missing something?

Reply:
I mean the other way round: you re-export `datawizard::add_labels()` as `sjlabelled::add_labels()`. 
But the hidden assumption here is that their behaviour are comparable.

Reply:
No. `datawizard::add_labels()` is more like `sjlabelled::set_label()` and `sjlabelled::set_labels()` (i.e. it covers both cases).

`sjlabelled::add_labels()` does not drop existing value labels, so you can overwrite and add labels to an already labelled variable:

> add_labels() adds labels to the existing value labels of x, however, unlike [set_labels](https://strengejacke.github.io/sjlabelled/reference/set_labels.html), it does not remove labels that were not specified in labels. add_labels() also replaces existing value labels, but preserves the remaining labels.

Reply:
>but if I had to choose I'd go with assign_labels()

>  Let's go with apply_labels()?

😄

Reply:
I change my mind every day when the perfect name is not available 😭

Reply:
Don't worry, the first vote was from @etiennebacher, the second yours. So you both just have to come to an agreement.